openwisp_radius/tests/test_tasks.py

```diff
@@ -0,0 +1,79 @@
+from celery import Celery
+from celery.contrib.testing.worker import start_worker
+from django.contrib.auth import get_user_model
+
+from openwisp_radius import tasks
+
+from ..utils import load_model
+from . import _RADACCT, FileMixin
+from .mixins import BaseTestCase
+
+User = get_user_model()
+RadiusAccounting = load_model('RadiusAccounting')
+RadiusBatch = load_model('RadiusBatch')
+RadiusPostAuth = load_model('RadiusPostAuth')
+
+
+class TestCelery(FileMixin, BaseTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        app = Celery('openwisp2')
+        app.config_from_object('django.conf:settings', namespace='CELERY')
+        app.autodiscover_tasks()
+        cls.celery_worker = start_worker(app)
```

Comment:
The tests here done are a little different than the ones in other modules, in the celery docs, I read that ideally, celery should only call tasks and shouldn't process any data.

> A Celery task is much like a web view, in that it should only define how to perform the action in the context of being called as a task.

This makes celery testing not required in most modules, however, since I wanted to test the calling of celery tasks, I created an instance of celery from instance the test class to ensure that celery is in the same process as tests.

This helps me avoid mocking. 
With mocking, I couldn't have tested the calls rather I would be testing management command operations only which are tested in `test_management` anyway.

What do you think about this approach?

Reply:
Looks good, I think the closer we get to replicating real world usage, the better, so your effort here seems to go in the right direction, but I'll be able to confirm this to you only in the future once we try to replicate this in other modules and find out if it actually catches more bugs than the approach we follow currently. For the moment, it's good, let's try it.